Source/Dafny/Resolver.cs

```diff
@@ -3489,161 +3499,126 @@ public class ModuleBindings
+            char c = ((String) o)[0];
+            if ((int) c > MaxBV(ce.Type)) {
+              return null; // Argument out of range
             }
+            return new BigInteger(((string) o)[0]);
+          }
 
-            if ((ce.E.Type.IsNumericBased(Type.NumericPersuation.Int) || ce.E.Type.IsBitVectorType) &&
+          if ((ce.E.Type.IsNumericBased(Type.NumericPersuation.Int) || ce.E.Type.IsBitVectorType) &&
                 ce.Type.IsCharType) {
-              BigInteger b = (BigInteger) o;
-              if (b < BigInteger.Zero || b > new BigInteger(65535)) {
-                reporter.Error(MessageSource.Resolver, ce.E,
-                  prefix + ", argument out of range: " + b.ToString());
-                return null;
-              }
-
-              return ((char) (int) b).ToString();
+            BigInteger b = (BigInteger) o;
+            if (b < BigInteger.Zero || b > new BigInteger(65535)) {
+              return null; // Argument out of range
             }
+            return ((char) (int) b).ToString();
+          }
 
-            if (ce.E.Type.IsCharType &&
-                ce.Type.IsNumericBased(Type.NumericPersuation.Real)) {
-              if (ce.Type.IsRealType) return Basetypes.BigDec.FromInt(((string) o)[0]);
-              return null;
-            }
+          if (ce.E.Type.IsCharType &&
+              ce.Type.IsNumericBased(Type.NumericPersuation.Real)) {
+            if (ce.Type.IsRealType) return Basetypes.BigDec.FromInt(((string) o)[0]);
+            return null;
+          }
 
-            if (ce.E.Type.IsNumericBased(Type.NumericPersuation.Real) &&
+          if (ce.E.Type.IsNumericBased(Type.NumericPersuation.Real) &&
                 ce.Type.IsCharType) {
-              BigInteger ff, cc;
-              ((Basetypes.BigDec) o).FloorCeiling(out ff, out cc);
-              if (((Basetypes.BigDec) o) != Basetypes.BigDec.FromBigInt(ff)) {
-                reporter.Error(MessageSource.Resolver, ce.E,
-                  prefix + ", argument not an integer: " + ((Basetypes.BigDec) o).ToString());
-                return null;
-              }
-
-              if (ff < BigInteger.Zero || ff > new BigInteger(65535)) {
-                reporter.Error(MessageSource.Resolver, ce.E,
-                  prefix + ", argument out of range: " + ff.ToString());
-                return null;
-              }
-
-              return ((char) (int) ff).ToString();
+            ((Basetypes.BigDec) o).FloorCeiling(out var ff, out _);
+            if (((Basetypes.BigDec) o) != Basetypes.BigDec.FromBigInt(ff)) {
+              return null; // Argument not an integer
+            }
+            if (ff < BigInteger.Zero || ff > new BigInteger(65535)) {
+              return null; // Argument out of range
             }
+            return ((char) (int) ff).ToString();
           }
+
         } else if (e is SeqSelectExpr sse) {
           Object b = GetAnyConst(sse.Seq, consts);
```

Comment:
Below, you're using `b` as its value is a (possibly `null`) `string`. (For example, L3620 would otherwise be a null-dereference.) So, you may as well declare `b` to be of type `string`.
```suggestion
          var b = GetAnyConst(sse.Seq, consts) as string;
```

Reply:
Fixed